ci/long_running_tests/workloads/serve_scalability.py

```diff
@@ -0,0 +1,67 @@
+# A multi-node scalability test. We put an http proxy on the head node and spin
+# up e.g. 20 nodes and put a bunch of replicas on the cluster, and run a stress
+# test.
+
+import logging
+import time
+import subprocess
+import requests
+
+import ray
+from ray import serve
+from ray.serve import BackendConfig
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+ray.init(address="auto")
+client = serve.start()
+
+# These numbers need to correspond with the autoscaler config file.
+# The number of remote nodes in the autoscaler should upper bound
+# these because sometimes nodes fail to update.
+num_remote_nodes = 4
+head_node_cpus = 2
+num_remote_cpus = num_remote_nodes * head_node_cpus
+
+# Wait until the expected number of nodes have joined the cluster.
+while True:
+    num_nodes = len(ray.nodes())
+    logger.info("Waiting for nodes {}/{}".format(num_nodes,
+                                                 num_remote_nodes + 1))
+    if num_nodes >= num_remote_nodes + 1:
+        break
```

Comment:
Why do we have the +1 here?

Reply:
Good catch, now that the resource counting bug https://github.com/ray-project/ray/pull/11084 is fixed I don't need this for the test to run

Reply:
Actually ignore my last comment, I was thinking of something else.  The +1 is used in `test_many_actors.py` and so on, I think because num_remote_nodes corresponds to min_workers and max_workers in the cluster yaml, which doesn't include the head node.

Reply:
Following your suggestion I used easier-to-understand variable names, so this +1 won't appear any longer.